pwnlib/commandline/template.py

```diff
@@ -32,10 +32,72 @@
                                         os.path.join(printable_data_path, "templates", "pwnup.mako"))
 parser.add_argument('--no-auto', help='Do not automatically detect missing binaries', action='store_false', dest='auto')
 
+def get_docker_image_libraries():
+    """Tries to retrieve challenge libraries from a Docker image built from the Dockerfile in the current working directory.
+    
+    The libraries are retrieved by parsing the output of running ldd on /bin/sh.
+    If the Dockerfile does not have an ENTRYPOINT, assumes that the image is based on a jail.
+    In that case, assumes that the jail is pwn.red/jail which requires "--privileged" and places the child image in /srv,
+    therefore chroots into /srv before running ldd.
+    """
+    log.info("Extracting challenge libraries from Docker image...")
+    dockerfile = open("Dockerfile", "r").read()
+    is_jailed = re.search(r"^ENTRYPOINT", dockerfile, re.MULTILINE) is None
+    try:
+        image_sha = subprocess.check_output(["docker", "build", "-q", "."], shell=False).decode().strip()
+
+        if is_jailed:
+            ldd_command = ["/bin/sh", "-c", "chroot /srv /bin/sh -c 'ldd /bin/sh'"]
+        else:
+            ldd_command = ["-c", "ldd /bin/sh"]
+        ldd_output = subprocess.check_output([
+                "docker",
+                "run",
+                "--rm"
+                ] + (["--privileged"] if is_jailed else ["--entrypoint", "/bin/sh"]) + [
+                    image_sha,
+                ] + ldd_command,
+            shell=False
+        ).decode()
+        
+        libc, ld = None, None
+        libc_basename, ld_basename = None, None
+        for line in ldd_output.splitlines():
+            if "libc." in line:
+                libc = line.split("=> ", 1)[1].split(" ", 1)[0].strip()
+                libc_basename = libc.rsplit("/", 1)[1]
+            if "ld-" in line:
+                ld = line.split(" ", 1)[0].strip()
+                ld_basename = ld.rsplit("/", 1)[1]
+
+        if not (libc and ld):
+            return None, None
+
+        for filename, basename in zip([libc, ld], [libc_basename, ld_basename]):
+            if is_jailed:
+                cat_command = ["/bin/sh", "-c", "chroot /srv /bin/sh -c '/bin/cat %s'" % filename]
```

Comment:
Yep, I have considered it. 
I was thinking that if an exception is thrown after we run the container, we would need to either leave it running or wrap the whole function in a `try: ... except Exception` to make sure we clean it up, which I both wanted to avoid.
If you still think it would be better to go with `docker cp`, I can implement it that way.